lib/Twig/Extensions/Node/Trans.php

```diff
@@ -161,6 +316,41 @@ protected function getTransFunction($plural)
     {
         return $plural ? 'ngettext' : 'gettext';
     }
+
+    /**
+     * Checks that two nodes are similar. This includes deep comparing their
+     * child nodes. Two nodes are similar if they both share the same tags,
+     * attributes and their child nodes are also similar.
+     *
+     * This explicity leaves out the "lineno" attribute and the "specialVars"
+     * who in the context of the {%trans%} tag they must be equal.
+     *
+     * @param TwigNode $first  The first node, source of the comparison
+     * @param TwigNode $second The second node to compare against
+     *
+     * @return bool Returns True if both Nodes seems similar
+     */
+    protected function isNodeSimilar(\Twig_Node $first, \Twig_Node $second)
+    {
+        // First sign that the Nodes are not even similar. Fail fast...
+        if (($first->tag != $second->tag) || ($first->attributes != $second->attributes) || ($first->count() != $second->count())) {
```

Comment:
shouldn't it also compare the node classes ? Here, different node types would be considered similar

Reply:
There is a previous check that filters only instances of `GetAttrExpression`, `NameExpression`, `TempNameExpression` and `ConstantExpression`. All other classes throw a `Twig_Error_Syntax` as the expression is "too complex" to resolve into a unique variable name.

Given that, the next lines I think guarantees a proper check by recursively looking mismatches into inner nodes. The first `if` only looks for a quick negative match so that we can shortcut a deep compare.